### ISSUE TITLE
require('react') may resolve to react-slick's local bower_components/react

### ISSUE THREAD

**Issue by grrowl** [akiran/react-slick#24]
in my webpack.config.js I have:
```js
  resolve: {
    extensions: ['', '.js', '.jsx'],
    modulesDirectories: ['bower_components', 'node_modules'],
  },
  plugins: [
    new webpack.ResolverPlugin(
      new webpack.ResolverPlugin.DirectoryDescriptionFilePlugin('bower.json', ['main'])
    ),
```

Which means the `require('react');` within react-slick.js resolves `react` to the react.js in its `bower_components` directory, and fails within React's `findComponentRoot` on (compiled) line 11563: `firstChildren[0] = deepestAncestor.firstChild;`.

The issue is that when -slick intitialises, React looks for the root node, which is generated by a *different* React, therefore `ReactMount.findComponentRoot` can't find `reactRoot`.

Deleting the `bower_components` directory inside `node_modules/react-slick` fixed my issue for now, but it will come back when a fresh environment does a `npm install` (including elastic beanstalk or heroku builds). I haven't got a permanent solution yet.

**Comment by grrowl** [comment]
**tl;dr:** The existence of `bower_components` in the `node_modules/react-slick/` directory may affect webpack's module resolution and use the bundled `React`, which is different from the one which rendered the page, causing a fatal error.

**Comment by ryanbagwell** [comment]
I'm having a similar issue. package.json has a postistall script, which is trying to call "bower install," except bower isn't listed as a dependency.

I'm wondering why you need bower at all here? Can't you just include all of the bower packages in package.json?

**Comment by akiran** [comment]
removed the dependency on bower and added slick-carousel as a npm dependency.

Released 0.3.7 on npm with change.
Check if your issues got resolved.

**Comment by ryanbagwell** [comment]
@akiran this worked great. Thanks for the fix and thanks for the app.

**Comment by grrowl** [comment]
:+1: works for me!
